.github/ISSUE_TEMPLATE/add_dataset.yml

```
@@ -0,0 +1,79 @@
+name: Add dataset
+description: Suggest a new Library, Archive or Museum dataset to include in the Hugging Face Hub.
+title: "[Add Dataset]: "
```

What about suggesting the the user proposes a dataset name?
```suggestion
title: "Add dataset: [dataset_name_in_snake_case]"
```

I would also propose to align with Datasets and naming it with snake_case (all in lowercase and word separated by underscores).
The validation step could also check this and rename the dataset if needed.